index.js

```diff
@@ -35,7 +35,7 @@ BitField.prototype.get = function(i){
 BitField.prototype.set = function(i, b){
 	var j = i >> 3;
 	if (b || arguments.length === 1){
-		this._grow(j + 1);
+		if (this.buffer.length < j + 1) this._grow(Math.max(j + 1, Math.min(2 * j, this.grow)));

```

Comment:
I think `2 * j` actually needs to be `2 * this.buffer.length`. Otherwise, you're not doubling the existing buffer size, but resizing to double whatever index the user just called `set` with.